From 94238cddf43d87ff9221eda79ea2d26b6f63138c Mon Sep 17 00:00:00 2001
From: Yongjun Zhang <yzhang@cloudera.com>
Date: Fri, 4 Dec 2015 13:45:01 -0800
Subject: [PATCH 0995/1023] HDFS-9474. TestPipelinesFailover should not fail
 when printing debug message. (John Zhuge via
 Yongjun Zhang)

(cherry picked from commit 59dbe8b3e96d13c2322cabd87c7f893c5a3812ba)

Change-Id: I8cc4b3d64db56d769189e38c500b25eb2cd64282
(cherry picked from commit 5e114aaf2a052998165176fb4fc9c71ab8e4726b)
(cherry picked from commit 6c559e1ca8b4d59faa335966dcaa3d27b3c325ea)
---
 .../server/namenode/ha/TestPipelinesFailover.java  |   38 ++++++++++----------
 1 file changed, 18 insertions(+), 20 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/ha/TestPipelinesFailover.java b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/ha/TestPipelinesFailover.java
index 7616d7f..e71d0d9 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/ha/TestPipelinesFailover.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/ha/TestPipelinesFailover.java
@@ -59,7 +59,7 @@
 import org.apache.hadoop.test.GenericTestUtils.DelayAnswer;
 import org.apache.hadoop.test.MultithreadedTestUtil.RepeatingTestThread;
 import org.apache.hadoop.test.MultithreadedTestUtil.TestContext;
-import org.apache.hadoop.util.Shell;
+import org.apache.hadoop.util.StringUtils;
 import org.apache.hadoop.util.Shell.ShellCommandExecutor;
 import org.apache.log4j.Level;
 import org.apache.log4j.Logger;
@@ -425,29 +425,27 @@ public void testPipelineRecoveryStress() throws Exception {
     // The following section of code is to help debug HDFS-6694 about
     // this test that fails from time to time due to "too many open files".
     //
+    LOG.info("HDFS-6694 Debug Data BEGIN");
 
-    // Only collect debug data on these OSes.
-    if (Shell.LINUX || Shell.SOLARIS || Shell.MAC) {
-      System.out.println("HDFS-6694 Debug Data BEGIN===");
-
-      String[] scmd = new String[] {"/bin/sh", "-c", "ulimit -a"};
-      ShellCommandExecutor sce = new ShellCommandExecutor(scmd);
-      sce.execute();
-      System.out.println("'ulimit -a' output:\n" + sce.getOutput());
-
-      scmd = new String[] {"hostname"};
-      sce = new ShellCommandExecutor(scmd);
-      sce.execute();
-      System.out.println("'hostname' output:\n" + sce.getOutput());
-
-      scmd = new String[] {"ifconfig", "-a"};
-      sce = new ShellCommandExecutor(scmd);
-      sce.execute();
-      System.out.println("'ifconfig' output:\n" + sce.getOutput());
+    String[][] scmds = new String[][] {
+      {"/bin/sh", "-c", "ulimit -a"},
+      {"hostname"},
+      {"ifconfig", "-a"}
+    };
 
-      System.out.println("===HDFS-6694 Debug Data END");
+    for (String[] scmd: scmds) {
+      String scmd_str = StringUtils.join(" ", scmd);
+      try {
+        ShellCommandExecutor sce = new ShellCommandExecutor(scmd);
+        sce.execute();
+        LOG.info("'" + scmd_str + "' output:\n" + sce.getOutput());
+      } catch (IOException e) {
+        LOG.warn("Error when running '" + scmd_str + "'", e);
+      }
     }
 
+    LOG.info("HDFS-6694 Debug Data END");
+
     HAStressTestHarness harness = new HAStressTestHarness();
     // Disable permissions so that another user can recover the lease.
     harness.conf.setBoolean(
-- 
1.7.9.5

